### PR TITLE
Develop mobile tabs and file selection chips component

### PR DESCRIPTION
**Pull request by BoZhaoUTSC**
Fixes Issue #

- [ ] Patch: Bug Fix?
- [x] Minor: New Feature?
- [ ] Major: Breaking Change?
- [ ] Documentation?
- [ ] New dependencies?

### Other Notes
- add peer selection tabs
- add file selection chips

- TODO: need to think about the file structure and how to pass it to tabs component

### CODE REVIEW

File: packages/client/components/peer-tabs/index.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      {children}
+    </Typography>
+  )
+}
+
+interface PeerTabsProps {
+  content: any[]
+}
+
+export const PeerTabs = ({ content = [] }: PeerTabsProps) => {
```
[the remark this answers is not in the document]
Comment (Mystaken) [reply packages/client/components/peer-tabs/index.tsx]:
```
export interface SwipeableTabViewsContent {
  label: string
}
export interface SwipeableTabViewsProps<T extends SwipeableTabViewsContent> {
  content: T[]
  viewRenderer: (content: T, index: number) => JSX.Element | null
  allView?: JSX.Element
}

export const SwipeableTabViews: <T extends SwipeableTabViewsContent>(
  props: SwipeableTabViewsProps<T>
) => JSX.Element | null = ({ content, viewRenderer, allView }) => {
  return (
    <Container>
      <TabsContainer>
        <Tabs
          >
          {content.map(({ label }) => (
            <Tab
              label={label}
            />
          ))}
        </Tabs>
      </TabsContainer>
      <SwipeableViews
      // ...props, etc
      >
        {content.map(viewRenderer)}
        {allView}
      </SwipeableViews>
    </Container>
  )
}
```

A rough idea

File: packages/client/components/peer-tabs/index.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  width: 100%;
+  position: relative;
+  min-height: 200px;
+`
+
+const TabsContainer = styled.div`
+  background-color: #f9f9f9;
+`
+
+interface TabPanelProps {
```
Comment (Mystaken) [inline packages/client/components/peer-tabs/index.tsx]: Let's export the props.

File: packages/client/components/peer-tabs/index.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      hidden={selectedTabIndex !== index}
+      id={`action-tabpanel-${index}`}
+      aria-labelledby={`action-tab-${index}`}>
+      {children}
+    </Typography>
+  )
+}
+
+interface PeerTabsProps {
+  content: any[]
```
[the remark this answers is not in the document]
Comment (BoZhaoUTSC) [reply packages/client/components/peer-tabs/index.tsx]: > If we only care about the length of the array, can just input the # of peers as a `number`?

We actually need to pass files and annotations into `PeerTabs` component. I didn't figure out what's the data structure yet. Will come back and modify this later.

File: packages/client/components/peer-tabs/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (BoZhaoUTSC) [reply packages/client/components/peer-tabs/index.tsx]: > ```
> export interface SwipeableTabViewsContent {
>   label: string
> }
> export interface SwipeableTabViewsProps<T extends SwipeableTabViewsContent> {
>   content: T[]
>   viewRenderer: (content: T, index: number) => JSX.Element | null
>   allView?: JSX.Element
> }
> 
> export const SwipeableTabViews: <T extends SwipeableTabViewsContent>(
>   props: SwipeableTabViewsProps<T>
> ) => JSX.Element | null = ({ content, viewRenderer, allView }) => {
>   return (
>     <Container>
>       <TabsContainer>
>         <Tabs
>           >
>           {content.map(({ label }) => (
>             <Tab
>               label={label}
>             />
>           ))}
>         </Tabs>
>       </TabsContainer>
>       <SwipeableViews
>       // ...props, etc
>       >
>         {content.map(viewRenderer)}
>         {allView}
>       </SwipeableViews>
>     </Container>
>   )
> }
> ```
> 
> A rough idea

The `PeersTab` is most likely the only tabs component.

File: packages/client/components/peer-tabs/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Mystaken) [reply packages/client/components/peer-tabs/index.tsx]: Ok, in the future, let's avoid `any` in favor of `unknown`. The latter is more type safe.

File: packages/client/components/peer-tabs/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mystaken) [reply packages/client/components/peer-tabs/index.tsx]: Alright.

**Pull request merged**